api/services/auth.js

```diff
@@ -17,109 +20,53 @@ const socialLogin = async data => {
   return jwtHelpers.generateToken(data);
 };
 
-/**
- * TODO: It needs to rebuild process of creating user Instance with Transactions
- *  https://sequelize.org/master/manual/transactions.htmlF
- */
+const registration = async ({ firstName, lastName, email, password }) => {
+  const user = await db.Users.findOne({ where: { email } });
+  if (user) {
+    throw new errors.UserAlreadyExistsError();
+  }
+
+  let transaction;
+  try {
+    transaction = await db.sequelize.transaction();
```

Comment:
If you wrapped ` transaction = await db.sequelize.transaction();` in try/catch
Than wrap into try/catch `const user = await db.Users.findOne({ where: { email } });`
Or push out from try/catch the assingment.